src/gluonts/ev/api.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from collections import ChainMap
+from dataclasses import dataclass, field
+from typing import (
+    Callable,
+    Collection,
+    Dict,
+    Iterator,
+    Optional,
+    Protocol,
+    runtime_checkable,
+)
+
+import numpy as np
+
+from gluonts.dataset.split import TestData
+from gluonts.time_feature.seasonality import get_seasonality
+from gluonts.model.predictor import Predictor
+from .stats import seasonal_error
+from .aggregations import Aggregation
+
+
+@runtime_checkable
+class Metric(Protocol):
+    def __call__(self, axis: Optional[int] = None) -> "MetricEvaluator":
+        raise NotImplementedError
+
+
+class EvaluationDataBatch:
+    """Used to add batch dimension
+    Should be replaced by a `ForecastBatch` eventually"""
+
+    def __init__(self, values) -> None:
+        self.values = values
+
+    def __getitem__(self, name):
+        return np.array([self.values[name]])
+
+
+def construct_data(
+    test_data: TestData, predictor: Predictor, **predictor_kwargs
+) -> Iterator[Dict[str, np.ndarray]]:
```

Comment:
This should not be part of `ev`. Let us have `ev` as an independent module, which does not know about anything in GluonTS.

Reply:
If we want to to have a method`evaluate(test_data, predictor)` in `MetricEvaluator`, we need to be able to deal with what the predictor gives us - `Forecast` objects.

To separate things, the only way I see is changing the interface to something like `evaluate(data: Iterator[Dict[str, np.ndarray]])`. Then, the user would have to call one module to prepare the data and another to evaluate it.

Also, the `data` needed to update the `MetricEvaluator`s use `Forecast` objects under the hood and the `ChainMap` just combines them.

So, I'm not sure if we can truly separate things and if not, if we should move the data preparation part to a different module.

Reply:
That's the thing, I don't want the method in `ev`!